venter/views.py

```diff
@@ -77,17 +78,33 @@ def create(cls, request):
             Complaints.objects.get(id=complaint.id)
         ).data, status=201)
 
-    def update(self, request, pk):
+    @login_required_ajax
+    def up_vote(self, request, pk):
+        """UpVote or un-UpVote a complaint {?action}=0,1"""
+
         complaint = self.get_complaint(pk)
-        complaint.users_up_voted.add(self.request.user.profile)
+
+        value = request.GET.get("action")
+        if value is None:
+            return Response({"message": "{?action} is required"}, status=400)
+
+        # Check possible actions
+        if value == "0":
+            complaint.users_up_voted.remove(self.request.user.profile)
+        elif value == "1":
+            complaint.users_up_voted.add(self.request.user.profile)
+        else:
+            return Response({"message": "Invalid Action"}, status=400)
+
         return Response(ComplaintSerializer(
             Complaints.objects.get(id=complaint.id)
-        ).data)
+        ).data, status=200)
```

Comment:
I'll let this pass, but you don't really need to specify this explicitly, since the default status code id `200` anyway.

Reply:
So should I change?

Reply:
Your call

Reply:
*just the couple of whitespace changes.

Reply:
Yes, I already marked that resolved.

Reply:
Check the tests.py file once again

Reply:
That's what I meant, nevermind